src/libs/Navigation/Navigation.ts

```diff
@@ -307,57 +284,53 @@ function setIsNavigationReady() {
 /**
  * Checks if the navigation state contains routes that are protected (over the auth wall).
  *
- * @function
- * @param {Object} state - react-navigation state object
- *
- * @returns {Boolean}
+ * @param state - react-navigation state object
  */
-function navContainsProtectedRoutes(state) {
-    if (!state || !state.routeNames || !_.isArray(state.routeNames)) {
+function navContainsProtectedRoutes(state: NavigationState | undefined): boolean {
+    if (!state?.routeNames || !Array.isArray(state.routeNames)) {
         return false;
     }
 
-    const protectedScreensName = _.values(PROTECTED_SCREENS);
-    const difference = _.difference(protectedScreensName, state.routeNames);
+    const protectedScreensName = Object.values(PROTECTED_SCREENS);
+    const difference = protectedScreensName.filter((screen) => !state.routeNames.includes(screen));
```

Comment:
I think it would be much more intuitive here to use `.some` rather than `.filter` then coercing `.length` into a boolean

Reply:
+1, adjusted